lti_consumer/lti_xblock.py

```diff
@@ -1506,17 +1572,23 @@ def _get_context_for_template(self):
         Returns:
             dict: Context variables for templates
         """
-
         # For more context on ALLOWED_TAGS and ALLOWED_ATTRIBUTES
         # Look into this documentation URL see https://bleach.readthedocs.io/en/latest/clean.html#allowed-tags-tags
         # This lets all plaintext through.
         allowed_tags = bleach.sanitizer.ALLOWED_TAGS + ['img']
         allowed_attributes = dict(bleach.sanitizer.ALLOWED_ATTRIBUTES, **{'img': ['src', 'alt']})
         sanitized_comment = bleach.clean(self.score_comment, tags=allowed_tags, attributes=allowed_attributes)
+
         launch_url = self.launch_url
+        lti_1p3_launch_url = self.lti_1p3_launch_url.strip()
+
+        lti_consumer = None
+        if self.config_type in ("external", "database"):
+            lti_consumer = self._get_lti_consumer()
 
-        if self.config_type == "external":
-            launch_url = self._get_lti_consumer().lti_launch_url
+        # The lti_launch_url property only exists on the LtiConsumer1p1.
+        if self.config_type == "external" or (self.config_type == "database" and self.lti_version == "lti_1p1"):
+            launch_url = lti_consumer.lti_launch_url
 
         # As we currently support only LTI 1.1 for external configuration, let's simply
         # use the same logic as the LTI 1.1 for getting the launch handler.
```

Comment:
the code below looks pretty suspect in that it will do some stuff with version and external just like above

maybe we can compact this together and do the boolean logic less often and set more variables in each branch